### PR TITLE
[Seq] Use Properties for inherent attributes.

### PR DESCRIPTION
**Pull request by dtzSiFive**

### CODE REVIEW

File: include/circt/Dialect/Seq/SeqOps.h [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #ifndef CIRCT_DIALECT_SEQ_SEQOPS_H
 #define CIRCT_DIALECT_SEQ_SEQOPS_H
 
+#include "mlir/Bytecode/BytecodeOpInterface.h"
```
Comment (teqdruid) [inline include/circt/Dialect/Seq/SeqOps.h:16]: Why is this required? Odd.

**Review by teqdruid — APPROVED** [review]
Thanks!

File: include/circt/Dialect/Seq/SeqOps.h (hunk 1 above)
Comment (dtzSiFive) [reply include/circt/Dialect/Seq/SeqOps.h:16]: Odd indeed.  I don't know, seems like a bug to be honest, maybe they're missing a header themselves.
